header/p2p/peer_tracker_test.go

```diff
@@ -0,0 +1,28 @@
+package p2p
+
+import (
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPeerTracker_GC(t *testing.T) {
+	h := createMocknet(t, 1)
+	p := newPeerTracker(h[0], time.Millisecond*200, time.Millisecond*1, 1, 5)
+	pid1 := peer.ID("peer1")
+	pid2 := peer.ID("peer2")
+	p.connectedPeers[pid1] = &peerStat{peerID: pid1, peerScore: 0.5}
+	p.disconnectedPeers[pid2] = &peerStat{peerID: pid2, removedAt: time.Now()}
```

Comment:
please also add 2 examples of peers that shouldn't be removed, and assert that the correct one has been removed